src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -7,8 +7,32 @@
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
-    @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException(
+                    "Login cannot be null");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException(
+                    "User with this login already exists");
+        }
+
+        if (user.getLogin().length() < 6) {
```

Comment:
You have a check for the minimum length of the login, which is good. However, there is no check for the maximum length of the login. It's always a good idea to limit the length of a string to prevent very long strings from being stored and to protect against attacks.